x-pack/plugin/eql/qa/correctness/src/javaRestTest/java/org/elasticsearch/xpack/eql/EsEQLCorrectnessIT.java

```diff
@@ -0,0 +1,187 @@
+    public void checkSearchContent() throws Exception {
+        assertNoSearchContexts(client());
+    }
+
+    @AfterClass
+    public static void wipeTestData() throws IOException {
+        try {
+            adminClient().performRequest(new Request("DELETE", "/*"));
+        } catch (ResponseException e) {
+            // 404 here just means we had no indexes
+            if (e.getResponse().getStatusLine().getStatusCode() != 404) {
+                throw e;
+            }
+        }
+    }
+
+    @Override
+    protected boolean preserveClusterUponCompletion() {
+        // Need to preserve data between parameterized tests runs
+        return true;
+    }
+
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOException {
+        RestClientBuilder builder = RestClient.builder(hosts);
+        configureClient(builder, settings);
+        builder.setRequestConfigCallback(
+            requestConfigBuilder -> requestConfigBuilder.setConnectTimeout(30000000)
+                .setConnectionRequestTimeout(30000000)
+                .setSocketTimeout(30000000)
+        );
+        builder.setStrictDeprecationMode(true);
+        return builder.build();
+    }
+
+    private EqlSpec spec;
+
+    public EsEQLCorrectnessIT(EqlSpec spec) {
+        this.spec = spec;
+    }
+
+    private RestHighLevelClient highLevelClient() {
+        if (highLevelClient == null) {
+            highLevelClient = new RestHighLevelClient(client(), ignore -> {}, Collections.emptyList()) {
+            };
+        }
+        return highLevelClient;
+    }
+
+    @ParametersFactory(shuffle = false, argumentFormatting = PARAM_FORMATTING)
+    public static Iterable<Object[]> parameters() throws Exception {
+        Collection<EqlSpec> specs;
+        try (InputStream is = EsEQLCorrectnessIT.class.getClassLoader().getResourceAsStream(QUERIES_FILENAME)) {
+            specs = EqlSpecLoader.readFromStream(is);
+        }
+        assertFalse("Found 0 queries for testing", specs.isEmpty());
+
+        List<Object[]> params = new ArrayList<>(specs.size());
+        for (EqlSpec spec : specs) {
+            params.add(new Object[] { spec });
+        }
+        return params;
+    }
+
+    public void test() throws Exception {
+        long totalTime = 0;
+        int queryNo = spec.queryNo();
+        /* For debugging
+        for (int i = 0; i < spec.filters().length; i++) {
+            String filterQuery = spec.filters()[i];
+            EqlSearchRequest eqlSearchRequest = new EqlSearchRequest(INDEX_NAME, filterQuery);
+            eqlSearchRequest.eventCategoryField("event_type");
+            eqlSearchRequest.size(100000);
+            EqlSearchResponse response = client.eql().search(eqlSearchRequest, commonRequestOptions);
+            assertEquals("Failed to match filter counts for query No: " + queryNo + " filterCount: " + i,
+                    spec.filterCounts()[i], response.hits().events().size());
+        } */
```

Comment:
This should be moved into a separate method and be guarded by an if statement for a bool flag which can be turned on/off if needed.